dashboard.el

```diff
@@ -407,55 +424,65 @@ Optional argument ARGS adviced function arguments."
         (forward-line 1)))
     max-line-length))
 
+(defun dashboard-insert-items ()
+  "Function to insert dashboard items.
+See `dashboard-item-generators' for all items available."
+  (let ((recentf-is-on (recentf-enabled-p))
+        (origial-recentf-list recentf-list)
+        (max-line-length 0))
+    (mapc (lambda (els)
+            (let* ((el (or (car-safe els) els))
+                   (list-size
+                    (or (cdr-safe els)
+                        dashboard-items-default-length))
+                   (item-generator
+                    (cdr-safe (assoc el dashboard-item-generators))))
+
+              (push (point) dashboard--section-starts)
+              (funcall item-generator list-size)
+              (goto-char (point-max))
+              
+              (when recentf-is-on
+                (setq recentf-list origial-recentf-list))
+              (setq max-line-length
+                    (max max-line-length (dashboard-maximum-section-length)))))
+          dashboard-items)
+
+    (when dashboard-center-content
+      (dashboard-center-text
+       (if dashboard--section-starts
+           (car (last dashboard--section-starts))
+         (point))
+       (point-max)))
+    (dashboard-insert-page-break)))
+
 (defun dashboard-insert-startupify-lists ()
   "Insert the list of widgets into the buffer."
   (interactive)
   (let ((inhibit-redisplay t)
         (recentf-is-on (recentf-enabled-p))
         (origial-recentf-list recentf-list)
-        (dashboard-num-recents (or (cdr (assoc 'recents dashboard-items)) 0))
-        (max-line-length 0))
+        (dashboard-num-recents (or (cdr (assoc 'recents dashboard-items)) 0)))
     (when recentf-is-on
       (setq recentf-list (dashboard-subseq recentf-list dashboard-num-recents)))
     (dashboard--with-buffer
+     (if (fboundp 'solaire-mode)
+         (if dashboard-use-solaire
+             (solaire-mode t)
+           (solaire-mode -1)))
```

Comment:
`solaire-mode` shouldn't be in this package. Users can achieve this by adding a hook to `dashboard-after-initialize-hook`.